x-pack/plugins/security_solution/server/endpoint/lib/response_actions/complete_external_actions_task.ts

```diff
@@ -0,0 +1,144 @@
+  taskManager: TaskManagerSetupContract;
+}
+
+export interface CompleteExternalResponseActionsTaskStartOptions {
+  taskManager: TaskManagerStartContract;
+  esClient: ElasticsearchClient;
+}
+
+export class CompleteExternalResponseActionsTask {
+  private wasSetup = false;
+  private wasStarted = false;
+  private log: Logger;
+  private esClient: ElasticsearchClient | undefined = undefined;
+  private cleanup: (() => void | Promise<void>) | undefined;
+  private taskTimeout = '20m'; // Default. Real value comes from server config
+  private taskInterval = '60s'; // Default. Real value comes from server config
+
+  constructor(protected readonly options: CompleteExternalResponseActionsTaskConstructorOptions) {
+    this.log = this.options.endpointAppContext.logFactory.get(
+      COMPLETE_EXTERNAL_RESPONSE_ACTIONS_TASK_TYPE
+    );
+  }
+
+  public async setup({ taskManager }: CompleteExternalResponseActionsTaskSetupOptions) {
+    if (this.wasSetup) {
+      throw new Error(`Task has already been setup!`);
+    }
+
+    this.wasSetup = true;
+
+    if (!this.options.endpointAppContext.experimentalFeatures.responseActionsSentinelOneV2Enabled) {
+      this.log.info(
+        `responseActionsSentinelOneV2Enabled feature flag is disabled. Task [${COMPLETE_EXTERNAL_RESPONSE_ACTIONS_TASK_TYPE}] will NOT be registered!`
+      );
+      return;
+    }
+
+    this.taskInterval =
+      this.options.endpointAppContext.serverConfig.completeExternalResponseActionsTaskInterval ??
+      this.taskInterval;
+    this.taskTimeout =
+      this.options.endpointAppContext.serverConfig.completeExternalResponseActionsTaskTimeout ??
+      this.taskTimeout;
+
+    this.log.info(
+      `Registering task [${COMPLETE_EXTERNAL_RESPONSE_ACTIONS_TASK_TYPE}] with timeout of [${this.taskTimeout}] and run interval of [${this.taskInterval}]`
+    );
+
+    taskManager.registerTaskDefinitions({
+      [COMPLETE_EXTERNAL_RESPONSE_ACTIONS_TASK_TYPE]: {
+        title: COMPLETE_EXTERNAL_RESPONSE_ACTIONS_TASK_TITLE,
+        timeout: this.taskTimeout,
+        createTaskRunner: () => {
+          if (!this.esClient) {
+            throw new EndpointError(
+              `esClient not defined. Was [${this.constructor.name}.start()] called?`
+            );
+          }
+
+          return new CompleteExternalActionsTaskRunner(
+            this.options.endpointAppContext.service,
+            this.esClient,
+            this.taskInterval
+          );
+        },
+      },
+    });
+  }
+
+  public async start({ taskManager, esClient }: CompleteExternalResponseActionsTaskStartOptions) {
+    if (this.wasStarted) {
+      throw new Error('Task has already been started!');
+    }
+
+    this.wasStarted = true;
+    this.esClient = esClient;
+
+    if (!this.options.endpointAppContext.experimentalFeatures.responseActionsSentinelOneV2Enabled) {
+      return;
+    }
+
+    try {
+      await taskManager.ensureScheduled({
+        id: COMPLETE_EXTERNAL_RESPONSE_ACTIONS_TASK_TYPE,
```

Comment:
we should add a version to the id so we can check the [id to prevent an outdated version from running](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution_serverless/server/task_manager/usage_reporting_task.ts#L121).

Reply:
Oh yeah. I'll add that. I saw that in the Manifest Manager task as well - do you know what would trigger a difference in version number? would it be something that can happen during an upgrade?